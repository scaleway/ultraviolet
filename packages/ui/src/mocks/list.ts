export type MockData = {
  department: string
  description: string
  id: string
  name: string
  reference: string | number
}

export const generateData = (count?: number, prefix?: string): MockData[] =>
  new Array(count).map((_, index) => ({
    department: index % 3 === 0 ? `Front` : `Not Front`,
    description: `Fake message for row ${index}`,
    id: `${prefix ? `${prefix}-` : ''}${index}`,
    name: `Scaler ${prefix ? `${prefix}-` : ''}${index}`,
    reference: prefix ? `${prefix}-${index}` : index,
  }))

const listMock = { generateData }

export default listMock
