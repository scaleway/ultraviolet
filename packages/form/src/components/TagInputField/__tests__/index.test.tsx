import { renderHook, screen, waitFor } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithForm, renderWithTheme } from '@utils/test'
import { useForm } from 'react-hook-form'
import { describe, expect, test, vi } from 'vitest'
import { mockErrors } from '../../../mocks'
import { Form, Submit, TagInputField } from '../..'

const alpha = /^[a-zA-Z]*$/

describe('TagInputField', () => {
  test('should render correctly', () => {
    const { asFragment } = renderWithForm(
      <TagInputField name="test" placeholder="placeholder" />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly with regex', () => {
    const { asFragment } = renderWithForm(
      <>
        <TagInputField
          clearable
          label="Test"
          name="test"
          regex={[alpha]}
          required
        />
        <Submit>Test</Submit>
      </>,
      { defaultValues: { test: ['4'] } },
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should works with defaultValues', async () => {
    const onSubmit = vi.fn()
    const { result } = renderHook(() =>
      useForm<{ test: string[] }>({ defaultValues: { test: ['First'] } }),
    )

    const { asFragment } = renderWithTheme(
      <Form errors={mockErrors} methods={result.current} onSubmit={onSubmit}>
        <TagInputField clearable label="Test" name="test" required />
        <Submit>Submit</Submit>
      </Form>,
    )

    await userEvent.click(screen.getByText('Submit'))
    await waitFor(() => {
      expect(onSubmit).toHaveBeenCalledTimes(1)
    })
    expect(onSubmit.mock.calls[0][0]).toEqual({
      test: ['First'],
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
