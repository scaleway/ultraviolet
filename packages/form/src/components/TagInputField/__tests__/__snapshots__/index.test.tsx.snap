// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagInputField > should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  background-color: #ffffff;
  padding: calc(0.75rem - 1px) 1rem;
  cursor: text;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-2:focus-within {
  border-color: #792dd4;
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-2[data-success="true"] {
  border-color: #22674e;
}

.emotion-2[data-error="true"] {
  border-color: #b3144d;
}

.emotion-2:hover {
  border-color: #792dd4;
}

.emotion-2[data-readonly="true"] {
  border-color: #d9dadd;
  background: #f9f9fa;
}

.emotion-2[data-disabled="true"] {
  border-color: #e9eaeb;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-4 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 0.875rem;
  background: inherit;
  color: #3f4250;
  border: none;
  outline: none;
  height: 100%;
}

.emotion-6::-webkit-input-placeholder {
  color: #727683;
}

.emotion-6::-moz-placeholder {
  color: #727683;
}

.emotion-6:-ms-input-placeholder {
  color: #727683;
}

.emotion-6::placeholder {
  color: #727683;
}

.emotion-6[data-size="large"] {
  font-size: 1rem;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
    >
      <div
        class="emotion-0 emotion-1"
      >
        <div>
          <div
            class="emotion-2 emotion-3"
            data-disabled="false"
            data-error="false"
            data-readonly="false"
            data-success="false"
          >
            <div
              class="emotion-4 emotion-5"
            >
              <input
                class="emotion-6 emotion-7"
                data-size="large"
                id=":r0:"
                name="test"
                placeholder="placeholder"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`TagInputField > should render correctly with regex 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-2 {
  color: #222638;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  background-color: #ffffff;
  padding: calc(0.75rem - 1px) 1rem;
  cursor: text;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-4:focus-within {
  border-color: #792dd4;
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-4[data-success="true"] {
  border-color: #22674e;
}

.emotion-4[data-error="true"] {
  border-color: #b3144d;
}

.emotion-4:hover {
  border-color: #792dd4;
}

.emotion-4[data-readonly="true"] {
  border-color: #d9dadd;
  background: #f9f9fa;
}

.emotion-4[data-disabled="true"] {
  border-color: #e9eaeb;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-6 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.emotion-11 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

.emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 2rem;
  padding: 0 0.5rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
  border-radius: 0.25rem;
  box-sizing: border-box;
  width: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  paragraph-spacing: 0;
  text-case: none;
  background: none;
  border: none;
  color: #3f4250;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0.125rem;
}

.emotion-14:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-14:active {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-14:hover,
.emotion-14:active {
  background: #e9eaeb;
  color: #222638;
}

.emotion-16 {
  vertical-align: middle;
  fill: currentColor;
  height: 16px;
  width: 16px;
  min-width: 16px;
  min-height: 16px;
}

.emotion-16 .fillStroke {
  stroke: currentColor;
  fill: none;
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 0.875rem;
  background: inherit;
  color: #3f4250;
  border: none;
  outline: none;
  height: 100%;
}

.emotion-18::-webkit-input-placeholder {
  color: #727683;
}

.emotion-18::-moz-placeholder {
  color: #727683;
}

.emotion-18:-ms-input-placeholder {
  color: #727683;
}

.emotion-18::placeholder {
  color: #727683;
}

.emotion-18[data-size="large"] {
  font-size: 1rem;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
}

.emotion-22 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.5rem;
  padding: 0 0.25rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.25rem;
  border-radius: 0.25rem;
  box-sizing: border-box;
  width: 1.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  paragraph-spacing: 0;
  text-case: none;
  background: none;
  border: none;
  color: #3f4250;
}

.emotion-22:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-22:active {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-22:hover,
.emotion-22:active {
  background: #e9eaeb;
  color: #222638;
}

.emotion-26 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 3rem;
  padding: 0 1rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
  border-radius: 0.25rem;
  box-sizing: border-box;
  width: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: not-allowed;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  paragraph-spacing: 0;
  text-case: none;
  background: #8c40ef;
  border: none;
  color: #ffffff;
  background: #e5dbfd;
  color: #ffffff;
}

.emotion-26:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
    >
      <div
        class="emotion-0 emotion-1"
      >
        <label
          class="emotion-2 emotion-3"
          for=":r2:"
        >
          Test
        </label>
        <div>
          <div
            class="emotion-4 emotion-5"
            data-disabled="false"
            data-error="false"
            data-readonly="false"
            data-success="false"
          >
            <div
              class="emotion-6 emotion-7"
            >
              <span
                class="emotion-8 emotion-9"
              >
                <span
                  class="emotion-10 emotion-11 emotion-3"
                >
                  4
                </span>
                <button
                  aria-label="Close tag"
                  class="emotion-13 emotion-14 emotion-15"
                  data-testid="close-tag"
                  type="button"
                >
                  <svg
                    class="emotion-16 emotion-17"
                    viewBox="0 0 16 16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
                    />
                  </svg>
                </button>
              </span>
              <input
                class="emotion-18 emotion-19"
                data-size="large"
                id=":r2:"
                name="test"
                placeholder=""
                type="text"
                value=""
              />
            </div>
            <div
              class="emotion-20 emotion-21"
            >
              <button
                aria-label="clear value"
                class="emotion-22 emotion-15"
                type="button"
              >
                <svg
                  class="emotion-16 emotion-17"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <button
        class="emotion-26 emotion-27"
        disabled=""
        type="submit"
      >
        Test
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`TagInputField > should works with defaultValues 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.25rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-2 {
  color: #222638;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.5rem;
  background-color: #ffffff;
  padding: calc(0.75rem - 1px) 1rem;
  cursor: text;
  background: #ffffff;
  border: 1px solid #d9dadd;
  border-radius: 0.25rem;
}

.emotion-4:focus-within {
  border-color: #792dd4;
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-4[data-success="true"] {
  border-color: #22674e;
}

.emotion-4[data-error="true"] {
  border-color: #b3144d;
}

.emotion-4:hover {
  border-color: #792dd4;
}

.emotion-4[data-readonly="true"] {
  border-color: #d9dadd;
  background: #f9f9fa;
}

.emotion-4[data-disabled="true"] {
  border-color: #e9eaeb;
  background: #f3f3f4;
  cursor: not-allowed;
}

.emotion-6 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  white-space: nowrap;
  border-radius: 0.25rem;
  padding: 0 0.5rem;
  gap: 0.5rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 1.5rem;
  color: #3f4250;
  background: #ffffff;
  border: solid 1px #d9dadd;
}

.emotion-11 {
  font-size: 0.75rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: 14.5rem;
}

.emotion-14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 2rem;
  padding: 0 0.5rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
  border-radius: 0.25rem;
  box-sizing: border-box;
  width: 2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  paragraph-spacing: 0;
  text-case: none;
  background: none;
  border: none;
  color: #3f4250;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0.125rem;
}

.emotion-14:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-14:active {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-14:hover,
.emotion-14:active {
  background: #e9eaeb;
  color: #222638;
}

.emotion-16 {
  vertical-align: middle;
  fill: currentColor;
  height: 16px;
  width: 16px;
  min-width: 16px;
  min-height: 16px;
}

.emotion-16 .fillStroke {
  stroke: currentColor;
  fill: none;
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 0.875rem;
  background: inherit;
  color: #3f4250;
  border: none;
  outline: none;
  height: 100%;
}

.emotion-18::-webkit-input-placeholder {
  color: #727683;
}

.emotion-18::-moz-placeholder {
  color: #727683;
}

.emotion-18:-ms-input-placeholder {
  color: #727683;
}

.emotion-18::placeholder {
  color: #727683;
}

.emotion-18[data-size="large"] {
  font-size: 1rem;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
}

.emotion-22 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 1.5rem;
  padding: 0 0.25rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.25rem;
  border-radius: 0.25rem;
  box-sizing: border-box;
  width: 1.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 0.875rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.25rem;
  paragraph-spacing: 0;
  text-case: none;
  background: none;
  border: none;
  color: #3f4250;
}

.emotion-22:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-22:active {
  box-shadow: 0px 0px 0px 3px #151a2d5c;
}

.emotion-22:hover,
.emotion-22:active {
  background: #e9eaeb;
  color: #222638;
}

.emotion-26 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 3rem;
  padding: 0 1rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 0.5rem;
  border-radius: 0.25rem;
  box-sizing: border-box;
  width: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  outline-offset: 2px;
  white-space: nowrap;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  font-family: Inter,Asap,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  paragraph-spacing: 0;
  text-case: none;
  background: #8c40ef;
  border: none;
  color: #ffffff;
}

.emotion-26:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-26:active {
  box-shadow: 0px 0px 0px 3px #8c40ef40;
}

.emotion-26:hover,
.emotion-26:active {
  background: #792dd4;
  color: #f9f9fa;
}

<div
    data-testid="testing"
  >
    <form
      novalidate=""
    >
      <div
        class="emotion-0 emotion-1"
      >
        <label
          class="emotion-2 emotion-3"
          for=":ra:"
        >
          Test
        </label>
        <div>
          <div
            class="emotion-4 emotion-5"
            data-disabled="false"
            data-error="false"
            data-readonly="false"
            data-success="false"
          >
            <div
              class="emotion-6 emotion-7"
            >
              <span
                class="emotion-8 emotion-9"
              >
                <span
                  class="emotion-10 emotion-11 emotion-3"
                >
                  First
                </span>
                <button
                  aria-label="Close tag"
                  class="emotion-13 emotion-14 emotion-15"
                  data-testid="close-tag"
                  type="button"
                >
                  <svg
                    class="emotion-16 emotion-17"
                    viewBox="0 0 16 16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
                    />
                  </svg>
                </button>
              </span>
              <input
                class="emotion-18 emotion-19"
                data-size="large"
                id=":ra:"
                name="test"
                placeholder=""
                type="text"
                value=""
              />
            </div>
            <div
              class="emotion-20 emotion-21"
            >
              <button
                aria-label="clear value"
                class="emotion-22 emotion-15"
                type="button"
              >
                <svg
                  class="emotion-16 emotion-17"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.28 4.22a.75.75 0 0 0-1.06 1.06L6.94 8l-2.72 2.72a.75.75 0 1 0 1.06 1.06L8 9.06l2.72 2.72a.75.75 0 1 0 1.06-1.06L9.06 8l2.72-2.72a.75.75 0 0 0-1.06-1.06L8 6.94z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <button
        class="emotion-26 emotion-27"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
